File: Data Structures & Algorithms/depth-of-binary-tree/submission-5.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        // BFS
        
        // Queue<TreeNode> lvl = new LinkedList<>();

        // if(root == null){
        //     return 0;
        // }
        
        // lvl.add(root);
        // int nodes = 0;


        // while (!lvl.isEmpty()){
        //     nodes++;
        //     int iterate = lvl.size();
            
        //     for (int i = 0; i < iterate; i++){
        //         TreeNode c = lvl.remove();
        //         TreeNode l = c.left;
        //         TreeNode r = c.right;
        //         if (l != null){
        //             lvl.add(l);
        //         }
        //         if (r != null){
        //             lvl.add(r);
        //         }
        //     }
            
        // }

        // return nodes;

        // **** recursive 

        if(root == null){
            return 0;
        }
        
        return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-5.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        res = []

        q = deque()
        q.append([root,0])
        
        while q:
            root = q.popleft()
            if root[0]:
                if len(res) - 1 < root[1]:
                    res.append([])
                res[root[1]].append(root[0].val)

                q.append([root[0].left, root[1] + 1])
                q.append([root[0].right, root[1] + 1])
        return res
